src/NodeHandler.ts

```diff
@@ -0,0 +1,51 @@
+import { Evented } from '@dojo/core/Evented';
+import { VNodeProperties } from '@dojo/interfaces/vdom';
+import Map from '@dojo/shim/Map';
+import { NodeHandlerInterface } from './interfaces';
+
+/**
+ * Enum to identify the type of event.
+ * Listening to 'Projector' will notify when projector is created or updated
+ * Listening to 'Widget' will notifiy when widget root is created or updated
+ */
+export enum NodeEventType {
+	Projector = 'Projector',
+	Widget = 'Widget'
+}
+
+export class NodeHandler extends Evented implements NodeHandlerInterface {
```

Comment:
There are two things I've been struggling with using meta that this could perhaps help with. The first is doing a reverse lookup and the second is knowing what the currently rendered keys are. I need the reverse lookup for intersection meta where I have a single observer that receives multiple events. I don't have a good use case for knowing all the keys, but I could see it being potentially helpful/related to whatever we'd have to do for a reverse lookup.